src/mrb_capability.c

```diff
@@ -266,6 +294,87 @@ mrb_value mrb_cap_getgid(mrb_state *mrb, mrb_value self)
     return mrb_fixnum_value((mrb_int)getgid());
 }
 
+mrb_value mrb_file_cap_init(mrb_state *mrb, mrb_value self)
+{
+    char *path;
+    mrb_file_cap_context *cap_ctx = (mrb_file_cap_context *)mrb_malloc(mrb, sizeof(mrb_file_cap_context));
+
+    mrb_get_args(mrb, "z", &path);
+    cap_ctx->path = path;
+    cap_ctx->cap = cap_get_file(path);
+
+    mrb_iv_set(mrb
+        , self
+        , mrb_intern_cstr(mrb, "mrb_file_cap_context")
+        , mrb_obj_value(Data_Wrap_Struct(mrb
+            , mrb->object_class
+            , &mrb_file_cap_context_type
+            , (void *)cap_ctx)
+        )
+    );
+
+    return self;
+}
+
+mrb_value mrb_file_cap_path(mrb_state *mrb, mrb_value self)
+{
+    mrb_file_cap_context *cap_ctx = mrb_file_cap_get_context(mrb, self, "mrb_file_cap_context");
+    return mrb_str_new_cstr(mrb, cap_ctx->path);
+}
+
+mrb_value mrb_file_cap_set_file(mrb_state *mrb, mrb_value self)
+{
+    mrb_value ary;
+    mrb_int identify;
+    int i, ret;
+    mrb_file_cap_context *cap_ctx = mrb_file_cap_get_context(mrb, self, "mrb_file_cap_context");
+
+    if(cap_ctx->cap == NULL) {
+        cap_ctx->cap = cap_init();
+    }
+
+    mrb_get_args(mrb, "iA", &identify, &ary);
+    struct RArray *a = mrb_ary_ptr(ary);
+    int ncap = a->len;
+
+    for (i = 0; i < ncap; i++) {
+        cap_ctx->capval[i] = (cap_value_t)mrb_fixnum(a->ptr[i]);
+    }
+    cap_set_flag(cap_ctx->cap, identify, ncap, cap_ctx->capval, CAP_SET);
+
+    ret = cap_set_file(cap_ctx->path, cap_ctx->cap);
+    if(ret < 0){
+        mrb_sys_fail(mrb, "cap_set_file() failed");
+    }
+
+    mrb_iv_set(mrb
+        , self
+        , mrb_intern_cstr(mrb, "mrb_file_cap_context")
+        , mrb_obj_value(Data_Wrap_Struct(mrb
+            , mrb->object_class
+            , &mrb_file_cap_context_type
+            , (void *)cap_ctx)
+        )
+    );
+
+    return self;
+}
+
+mrb_value mrb_file_cap_to_text(mrb_state *mrb, mrb_value self)
+{
+    mrb_file_cap_context *cap_ctx = mrb_file_cap_get_context(mrb, self, "mrb_file_cap_context");
+    if(cap_ctx->cap == NULL) {
+        return mrb_str_new_cstr(mrb, "<Not yet set.>");
```

Comment:
You should use `mrb_str_new_lit` for literal. This function don't allocate buffer.